cmds/mount/mount.go

```diff
@@ -25,14 +28,38 @@ var (
 	opt    = flag.String("o", "", "Specify mount options")
 )
 
+func translateUnknownFS(originFS string, originErr error) error {
+	file, err := os.Open("/proc/filesystems")
+	if err != nil {
+		// just don't make things even worse...
+		return originErr
+	}
+	defer file.Close()
+	scanner := bufio.NewScanner(file)
+	for scanner.Scan() {
+		line := scanner.Text()
+		tokens := bufio.NewScanner(strings.NewReader(line))
+		tokens.Split(bufio.ScanWords)
+		var fs string
+		for tokens.Scan() {
+			fs = tokens.Text()
+		}
+		// just check the last token of the line
+		if fs == originFS {
+			return originErr
+		}
+	}
+	return errors.New("Unknown filesystem, check /proc/filesystems for supported ones")
+}
+
```

Comment:
See https://play.golang.org/p/ZPC0chiVOGV for a slightly less involved way to do this.
You can start with 
b, err := ioutil.ReadFile("/proc/filesystems")
if err != nil {
return
}
for s := range strings.Split(string(s), "\n")) {
...
}

Up to you, this code seems like  more work than you need. In any event, see the comment below.